db/compaction/compaction_picker_fifo.cc

```diff
@@ -443,9 +446,32 @@ Compaction* FIFOCompactionPicker::PickCompaction(
         cf_name, mutable_cf_options, mutable_db_options, vstorage, log_buffer);
   }
   RegisterCompaction(c);
+  UpdateCompactionStats(c);
   return c;
 }
 
+void FIFOCompactionPicker::UpdateCompactionStats(Compaction* c) {
+  if (c == nullptr || ioptions_.statistics == nullptr) {
+    return;
+  }
+
+  CompactionReason reason = c->compaction_reason();
+  Statistics* statistics = ioptions_.statistics.get();
+  switch (reason) {
+    case CompactionReason::kFIFOMaxSize:
+      RecordTick(statistics, FIFO_MAX_SIZE_COMPACTIONS);
+      break;
+    case CompactionReason::kFIFOTtl:
+      RecordTick(statistics, FIFO_TTL_COMPACTIONS);
+      break;
+    case CompactionReason::kChangeTemperature:
+      RecordTick(statistics, FIFO_CHANGE_TEMPERATURE_COMPACTIONS);
+      break;
+    default:
+      break;
+  }
```

Comment:
Based on the names I would have guessed only completed compactions count. Is it difficult to make it work that way?

Reply:
For deletion compaction (the scope of this PR), it's not difficult. Fixed. Thank you!